src/fontlibc/fontlibc.h

```diff
@@ -0,0 +1,437 @@
+/**
+ * @file
+ * @authors DrDnar
+ * @brief Provides improved font support
+ */
+
+#ifndef H_FONTLIBC
+#define H_FONTLIBC
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define FONTLIB_ENABLE_AUTO_WRAP 1
+#define FONTLIB_AUTO_CLEAR_TO_EOL 2
+#define FONTLIB_PRECLEAR_NEWLINE 4
+
+typedef enum {
+	/* clear = sans-serif font */
+	SERIF = 0x01,
+	/* If both are set, then assume there's no difference between oblique
+	 * and italic styles. */
+	OBLIQUE = 0x02,
+	ITALIC = 0x04,
+	/* Chances are you're not using this library for monospaced fonts.
+	 * But if you are, you'll still have to provide a widths table where
+	 * every byte is the same. */
+	MONOSPACED = 0x08
+} font_styles_t;
+
+typedef struct {
+	/* These are standard C-strings.  These may be NULL. */
+	intptr_t font_family_name;
+	intptr_t font_author;
+	/* NOTA BENE: TYPEFACES AND BITMAPPED FONTS CANNOT BE COPYRIGHTED UNDER U.S. LAW!
+	 * This field is therefore referred to as a pseudocopyright.  HOWEVER,
+	 * it IS is applicable in other jusrisdictions, such as Germany. */
+	intptr_t font_pseudocopyright;
+	intptr_t font_description;
+	intptr_t font_version;
+	intptr_t font_codepage;
+} font_metadata;
+
+typedef struct {
+	/* Must be zero */
+	uint8_t fontVersion;
+	/* Height in pixels */
+	uint8_t height;
+	/* Total number of glyphs provided. */
+	uint8_t total_glyphs;
+	/* Number of first glyph.  If you have no codepoints below 32, for
+	   example, you can omit the first 32 bitmaps. */
+	uint8_t first_glyph;
+	/* Offset/pointer to glyph widths table.
+	 * This is an OFFSET from the fontVersion member in data format.
+	 * However, it is 24-bits long because it becomes a real pointer upon loading. */
+	intptr_t widths_table;
+	/* Offset to a table of offsets to glyph bitmaps.
+	 * These offsets are only 16-bits each to save some space. */
+	intptr_t bitmaps;
+	/* Specifies how much to move the cursor left after each glyph.
+	   Total movement is width - overhang.  Intended for italics. */
+	uint8_t italic_space_adjust;
+	/* These suggest adding blank space above or below each line of text.
+	   This can increase legibility. */
+	uint8_t space_above;
+	uint8_t space_below;
+	/* Specifies the boldness of the font. 
+	   0x40: light
+	   0x80: regular
+	   0x90: medium
+	   0xC0: bold*/
+	uint8_t weight;
+	/* Specifies the style of the font.  See enum font_styles */
+	uint8_t style;
+	/* For layout, allows aligning text of differing fonts vertically.
+	   These count pixels going down, i.e. 0 means the top of the glyph. */
+	uint8_t cap_height;
+	uint8_t x_height;
+	uint8_t baseline_height;
+} raw_font;
+
+typedef struct {
+	char header[8]; /* "FONTPACK" */
+	/* Frankly, if you have more than 127 fonts in a pack, you have a
+	   problem. */
+	uint8_t fontCount;
+	/* Offset from first byte of header */
+	intptr_t metadata;
```

Comment:
Assuming this is a pointer to an array of pointers,

```suggestion
	fontlib_metadata_t **metadata;
```

Reply:
My intent was that a font pack would contain related fonts (i.e. a font family in various sizes and styles), and therefore only have one set of metadata.

Reply:
I have edited my original suggestion accordingly.

But in that case, where there's just one metadata per font pack, why is this a pointer at all? The metadata struct could just be embedded directly.